examples/sampling/demo_diffusion_sde.py

```diff
@@ -0,0 +1,539 @@
+
+This method requires:
+
+* A well-trained denoiser with varying noise levels (ideally with large noise levels) (e.g., :class:`deepinv.models.NCSNpp`).
+
+* A (noisy) data fidelity term (e.g., :class:`deepinv.sampling.DPSDataFidelity`).
+
+* Define a drift term :math:`f(x, t)` and a diffusion term :math:`g(t)` for the forward-time SDE. They can be defined through the :class:`deepinv.sampling.DiffusionSDE` (e.g., :class:`deepinv.sampling.VarianceExplodingDiffusion`).
+
+The :class:`deepinv.sampling.PosteriorDiffusion` class can be used to perform posterior sampling for inverse problems.
+Consider the acquisition model:
+
+.. math::
+     y = \noise{\forw{x}}.
+
+This class defines the reverse-time SDE for the posterior distribution :math:`p(x|y)` given the data :math:`y`:
+
+.. math::
+     d\, x_t = \left( f(x_t, t) - \frac{1 + \alpha}{2} g(t)^2 \nabla_{x_t} \log p_t(x_t | y) \right) d\,t + g(t) \sqrt{\alpha} d\, w_{t}
+
+where :math:`f` is the drift term, :math:`g` is the diffusion coefficient and :math:`w` is the standard Brownian motion. The drift term and the diffusion coefficient are defined by the underlying (unconditional) forward-time SDE `sde`. The (conditional) score function :math:`\nabla_{x_t} \log p_t(x_t | y)` can be decomposed using the Bayes' rule:
+
+.. math::
+     \nabla_{x_t} \log p_t(x_t | y) = \nabla_{x_t} \log p_t(x_t) + \nabla_{x_t} \log p_t(y | x_t).
+
+The first term is the score function of the unconditional SDE, which is typically approximated by a MMSE denoiser (`denoiser`) using the well-known Tweedie's formula, while the
+second term is approximated by the (noisy) data-fidelity term (`data_fidelity`).
+We implement various data-fidelity terms in :class:`deepinv.sampling.NoisyDataFidelity`.
+"""
+
+# %% Define the underlying SDE for posterior sampling
+# ---------------------------------------------------
+#
+# Let us import the necessary modules, define the denoiser and the SDE.
+#
+# In this example, we use the Variance-Exploding SDE, whose forward process is defined as:
+#
+# .. math::
+#     d\, x_t = \sigma(t) d\, w_t \quad \mbox{where } \sigma(t) = \sigma_{\mathrm{min}}\left( \frac{\sigma_{\mathrm{max}}}{\sigma_{\mathrm{min}}}\right)^t
+
+import torch
+import numpy as np
+
+import deepinv as dinv
+from deepinv.models import NCSNpp, EDMPrecond
+
+device = "cuda" if torch.cuda.is_available() else "cpu"
+dtype = torch.float64
+figsize = 2.5
+# %%
+from deepinv.sampling import (
+    PosteriorDiffusion,
+    DPSDataFidelity,
+    EulerSolver,
+    VarianceExplodingDiffusion,
+)
+from deepinv.optim import ZeroFidelity
+
+# In this example, we use the pre-trained FFHQ-64 model from the
+# EDM framework: https://arxiv.org/pdf/2206.00364 .
+# The network architecture is from Song et al: https://arxiv.org/abs/2011.13456 .
+unet = NCSNpp.from_pretrained("edm-ffhq64-uncond-ve")
```

Comment:
I would do like with the rest of the denoisers of the library 


```suggestion
unet = NCSNpp(pretrained="edm-ffhq64-uncond-ve")
```

Reply:
Yes this is a good remark, Thanks